config/settings/production.py

```diff
@@ -141,44 +142,61 @@
 
 # Your stuff...
 # ------------------------------------------------------------------------------
-# Extend middleware to add OIDC middleware
-MIDDLEWARE += ["mozilla_django_oidc.middleware.SessionRefresh"]  # noqa F405
-
-# Extend middleware to add OIDC auth backend
-AUTHENTICATION_BACKENDS += ["scram.route_manager.authentication_backends.ESnetAuthBackend"]  # noqa F405
-
-# https://docs.djangoproject.com/en/dev/ref/settings/#login-url
-LOGIN_URL = "oidc_authentication_init"
-
-# https://docs.djangoproject.com/en/dev/ref/settings/#login-redirect-url
-LOGIN_REDIRECT_URL = "/"
-
-# https://docs.djangoproject.com/en/dev/ref/settings/#logout-url
-LOGOUT_URL = "oidc_logout"
-
-# Need to point somewhere otherwise /oidc/logout/ redirects to /oidc/logout/None which 404s
-# https://github.com/mozilla/mozilla-django-oidc/issues/118
-# Using `/` because named urls don't work for this package
-# https://github.com/mozilla/mozilla-django-oidc/issues/434
-LOGOUT_REDIRECT_URL = "/"
-
-OIDC_OP_JWKS_ENDPOINT = os.environ.get(
-    "OIDC_OP_JWKS_ENDPOINT",
-    "https://example.com/auth/realms/example/protocol/openid-connect/certs",
-)
-OIDC_OP_AUTHORIZATION_ENDPOINT = os.environ.get(
-    "OIDC_OP_AUTHORIZATION_ENDPOINT",
-    "https://example.com/auth/realms/example/protocol/openid-connect/auth",
-)
-OIDC_OP_TOKEN_ENDPOINT = os.environ.get(
-    "OIDC_OP_TOKEN_ENDPOINT",
-    "https://example.com/auth/realms/example/protocol/openid-connect/token",
-)
-OIDC_OP_USER_ENDPOINT = os.environ.get(
-    "OIDC_OP_USER_ENDPOINT",
-    "https://example.com/auth/realms/example/protocol/openid-connect/userinfo",
-)
-OIDC_RP_SIGN_ALGO = "RS256"
-
-OIDC_RP_CLIENT_ID = os.environ.get("OIDC_RP_CLIENT_ID")
-OIDC_RP_CLIENT_SECRET = os.environ.get("OIDC_RP_CLIENT_SECRET")
+logging.info(f"Using AUTH METHOD = {AUTH_METHOD}")
```

Comment:
The changes to this file and to config/settings/test.py feel very repetitive.

I think this logic should simply stay in config/settings/base.py, it should rely on environment variables when things can be user-configurable, and it should rely on `if` statements with hardcoded values when things cannot be user-configurable (e.g. setting `LOGIN_URL = "oidc_authentication_init"` when oidc is enabled.

If we don't want to/can't add support for oidc to local.py, check for that setting and raise either a NotImplementedError if we plan to support it later, or a ValueError if that combination doesn't make sense.